numpy/distutils/ccompiler.py

```diff
@@ -136,20 +140,53 @@ def CCompiler_spawn(self, cmd, display=None):
         if is_sequence(display):
             display = ' '.join(list(display))
     log.info(display)
-    s, o = exec_command(cmd)
-    if s:
-        if is_sequence(cmd):
-            cmd = ' '.join(list(cmd))
-        try:
-            print(o)
-        except UnicodeError:
-            # When installing through pip, `o` can contain non-ascii chars
-            pass
-        if re.search('Too many open files', o):
-            msg = '\nTry rerunning setup command until build succeeds.'
+    try:
+        # cmd is documented to be a string and input
+        # such as 'gfortran --coverage' needs to be
+        # split
+
+        # the docs do not reflect the input received
+        # in practice though (often a list); process
+        # accordingly
+
+        if isinstance(cmd, np.compat.basestring):
+            # we do not reassign the value of cmd to a list
+            # here because of the is_sequence check below
+            subprocess.check_output(shlex.split(cmd,
+                                                posix=(os.name == 'posix')))
         else:
-            msg = ''
-        raise DistutilsExecError('Command "%s" failed with exit status %d%s' % (cmd, s, msg))
+            subprocess.check_output(cmd)
```

Comment:
Why can't we let `subprocess` do the splitting internally?

Reply:
I thought I had a good reason for doing that, but maybe not now that I've hacked things downstream in the control flow. Things seem to work locally if I remove the splitting block above, even with coverage flags exported in env.

I'm slightly nervous to make more changes now because of the extreme sensitivity to changes in this PR branch, but perhaps I could remove that split & squash / push.